cdap-ui/app/directives/complex-schema/complex-schema.less

```diff
@@ -168,68 +149,55 @@ my-complex-schema {
   }
 
   my-embedded-schema-selector {
-    > div { box-shadow: -7px 0 6px -6px rgba(1, 0, 0, 0.25); }
+    > div { position: relative; }
     [class*="col-"] {
       position: relative;
+      &:before {
+        box-shadow: -2px 0 3px 0 rgba(1, 0, 0, 0.25);
+        content: '';
+        display: block;
+        position: absolute;
+        top: -8px;
+        left: 0;
+        bottom: 0;
+        width: 1px;
+      }
       &:after {
-        box-shadow: 0 -3px 5px 0 rgba(1, 0, 0, 0.25);
+        box-shadow: 0 -2px 3px 0 rgba(1, 0, 0, 0.25);
         content: '';
         display: block;
         position: absolute;
         top: 0;
         right: 0;
-        height: 4px;
+        height: 1px;
         width: 155px;
       }
     }
     span.fa { color: @hydrator-blue; }
     div.tab-toggle {
-      position: relative;
-      top: -14px;
-      line-height: 10px;
-      height: 10px;
-      width: 20px;
-      text-align: center;
-      vertical-align: middle;
+      position: absolute;
+      z-index: 9999;
+      top: -15px;
+      left: 5px;
       &:hover,
       &:focus { cursor: pointer; }
     }
-    my-embedded-schema-selector {
-      div.tab-toggle {
-        top: -10px;
-      }
-      my-embedded-schema-selector {
-        div.tag-toggle {
-          top: -5px;
-        }
-      }
-    }
   }
 
-  my-union-schema {
-    .item > my-embedded-schema-selector > div > .clearfix > .col-xs-12:after {
-      width: ~"-moz-calc(100% - 85px)";
-      width: ~"-webkit-calc(100% - 85px)";
-      width: ~"calc(100% - 85px)";
-    }
+  my-embedded-schema-selector my-record-schema > my-complex-schema > .complex-schema-container {
+    .nested-row { border-top: 0; }
   }
+}
 
-  my-array-schema {
-    & > .array-schema-container > my-embedded-schema-selector > div > .clearfix > .col-xs-12:after {
-      width: ~"-moz-calc(100% - 85px)";
-      width: ~"-webkit-calc(100% - 85px)";
-      width: ~"calc(100% - 85px)";
-    }
-  }
+.column-content > my-complex-schema > .complex-schema-container > fieldset > div.field-row:not(.nested-row) {
+  box-shadow: -7px 0 6px -6px rgba(1, 0, 0, 0.25);
+  margin-left: 4px;
+}
 
-  my-map-schema {
-    & > .map-schema-container > my-embedded-schema-selector > div > .clearfix > .col-xs-12:after {
-      width: ~"-moz-calc(100% - 150px)";
-      width: ~"-webkit-calc(100% - 150px)";
-      width: ~"calc(100% - 150px)";
-    }
-  }
-  my-embedded-schema-selector my-record-schema .complex-schema-container {
-    .nested-row { border-top: 0; }
-  }
+.state-hydratorplusplus {
```

Comment:
I feel like these state specific styling should go inside their respective features' CSS instead of in the directive